src/Microsoft.DotNet.Darc/src/DarcLib/VirtualMonoRepo/VmrUpdater.cs

```diff
@@ -202,9 +202,27 @@ public class VmrUpdater : VmrManagerBase, IVmrUpdater
         }
         else
         {
+            // We squash commits and list them in the message
             var commitMessages = new StringBuilder();
+            var commitCount = 0;
+            bool commitLimitHit = false;
             while (commitsToCopy.TryPop(out LibGit2Sharp.Commit? commit))
             {
+                // Do not list over 23 commits in the message
+                // If there are more, list first 20
+                commitCount++;                
+                if (commitCount == 20 && commitsToCopy.Count > 3)
+                {
+                    commitLimitHit = true;
+                    commitMessages.AppendLine($"  [... commit list trimmed ...]");
+                    continue;
+                }
+
+                if (commitCount > 20 && commitLimitHit)
+                {
+                    continue;
```

Comment:
maybe just break, or have a bool flag in the while so we don't just spin in the while till we're out of commits?

Reply:
True, fixed! embarrassing 😅

Reply:
We could include the `&& !commitLimitHit` in the while condition

Reply:
Even better

Reply:
Check now